src/order/index.ts

```diff
@@ -0,0 +1,643 @@
+        appArgs: [
+          encodeString("create_application"),
+          intToBytes(this.registryAppId),
+          intToBytes(this.vaultAppId),
+          decodeAddress(this.userAddress).publicKey
+        ],
+        approvalProgram: base64ToBytes(APPROVAL_PROGRAM),
+        clearProgram: base64ToBytes(CLEAR_PROGRAM),
+        numGlobalByteSlices: ORDER_APP_GLOBAL_SCHEMA.numByteSlice,
+        numGlobalInts: ORDER_APP_GLOBAL_SCHEMA.numUint,
+        numLocalByteSlices: ORDER_APP_LOCAL_SCHEMA.numByteSlice,
+        numLocalInts: ORDER_APP_LOCAL_SCHEMA.numUint,
+        extraPages: ORDER_APP_EXTRA_PAGES
+      }),
+      algosdk.makeApplicationNoOpTxnFromObject({
+        sender: userAddress,
+        suggestedParams: sp,
+        appIndex: this.registryAppId,
+        appArgs: [encodeString("create_entry")],
+        boxes: [{appIndex: 0, name: entryBoxName}]
+      })
+    );
+
+    return this.setupTxnFeeAndAssignGroupId({txns: transactions});
+  }
+
+  async checkOrderAppAvailability(orderAppId?: number) {
+    if (!this.appId && !this.applicationAddress) {
+      if (orderAppId) {
+        this.appId = orderAppId;
+        this.applicationAddress = getApplicationAddress(orderAppId);
+      } else {
+        const applicationId = await OrderingClient.getOrderApplicationId(
+          this.algod,
+          this.network,
+          this.userAddress
+        );
+
+        if (applicationId) {
+          this.appId = Number(applicationId);
+          this.applicationAddress = getApplicationAddress(this.appId);
+        } else {
+          throw new Error(
+            "Ordering client is not found for this account address. Please create an order app first."
+          );
+        }
+      }
+    }
+  }
+
+  async getPutOrderTxnFee({
+    assetInId,
+    assetOutId,
+    type
+  }: {
+    assetInId: number;
+    assetOutId: number;
+    type: OrderType;
+  }) {
+    let totalFee = 0n;
+
+    // If the application ID is not set yet, ignore the fee calculation
+    if (!this.applicationAddress) {
+      return totalFee;
+    }
+
+    try {
+      const suggestedMinFee = (await this.getSuggestedParams()).minFee;
+      const assetsToOptin = await this.getAssetsToOptInToOrderingClient(
+        this.applicationAddress,
+        [assetInId, assetOutId]
+      );
+
+      const newOrderId = await this.getOrderCount();
+      const orderBoxName = this.getOrderBoxName(newOrderId, type);
+
+      // We can assume that the order box is not created with the new order id yet.
+      const newBoxes: Record<string, Struct> = {
+        [bytesToBase64(orderBoxName)]: ORDER_STRUCT
+      };
+
+      totalFee +=
+        BigInt(
+          this.calculateMinBalance({
+            boxes: newBoxes,
+            assets: assetsToOptin.length
+          })
+        ) + suggestedMinFee;
+
+      totalFee +=
+        // eslint-disable-next-line no-magic-numbers
+        BigInt(3 + assetsToOptin.length + Number(Boolean(assetsToOptin.length))) *
```

Comment:
It can be good idea to follow the magic numbers rule here. It is not clear what 3 comes from. Maybe we can add a comment explaining it, or define a variable with a descriptive name if possible